instrumentation.go

```diff
@@ -122,6 +129,11 @@ func (c instConfig) applyEnv() instConfig {
 	if v, ok := os.LookupEnv(envTargetExeKey); ok {
 		c.target = &process.TargetArgs{ExePath: v}
 	}
+	if v, ok := os.LookupEnv(envServiceNameKey); ok {
+		c.serviceName = v
+	} else if c.serviceName == "" {
+		c.serviceName = serviceNameDefault
```

Comment:
This needs to fallback to the executable name according to [the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/5f9bc69cebd2a45aa889d0a69ea9943b4257d85a/specification/resource/semantic_conventions/README.md#service). I.e.

https://github.com/open-telemetry/opentelemetry-go/blob/9bbefc6cc31995c7bdd37c69eb153fc2201c3562/sdk/resource/builtin.go#L101-L105